libsoftwarecontainer/src/softwarecontainer.cpp

```diff
@@ -165,45 +165,91 @@ void SoftwareContainer::addGateway(Gateway *gateway)
     m_gateways.push_back(std::move(std::unique_ptr<Gateway>(gateway)));
 }
 
-void SoftwareContainer::setGatewayConfigs(const GatewayConfiguration &gwConfig)
+ReturnCode SoftwareContainer::setGatewayConfigs(const GatewayConfiguration &gwConfig)
 {
-    /* TODO:
-     * The below assumption is incorrect, we do not want to appy configs to
-     * activated gateways.
-     */
-    // Go through the active gateways and check if there is a configuration for it
-    // If there is, apply it.
+    ReturnCode result = ReturnCode::SUCCESS;
 
+    result = configureGateways(gwConfig);
+    if (isError(result)) {
+        return result;
```

Comment:
I considered that at first. But then I started to use isError() to decide if the return is in fact something we should consider an error. I assume this check is available like this in order to have an abstraction between the check and the type, e.g. for extending it later and separating the logic of decision. So if the check is done with isError() I think it is more consistent with the abstraction to not at that point decide it means ReturnCode::FAILURE specifically. I don't see the benefit in isError() if we are not using it consistently. E.g. using it to enable extensibility and separation of concerns will break anyway if we interpret "error" as ReturnCode::FAILURE in the middle of the code.

If we however have some design principle about at what level of abstraction we decide to interpret "error" to ReturnCode::FAILURE specifically, then I can see a reason to do that in specific places, e.g. at some "end point" beyond which we use different abstractions.

If we should do this more explicit (which has its pros and cons as well of course), then I think the if should look for it explicitly as well, and we should re-consider the purpose of isError(), isSuccess() etc.